linkml_runtime/utils/schemaview.py

```diff
@@ -1292,6 +1293,35 @@ def slot_range_as_union(self, slot: SlotDefinition) -> List[EnumDefinitionName]:
                 range_union_of.append(x.range)
         return range_union_of
 
+    def get_classes_by_slot(self, slot: SlotDefinition, direct: bool, induced: bool) -> List[ClassDefinitionName]:
+        """Get all classes that use a given slot, either as a direct or induced slot.
+
+        :param slot: slot in consideration
+        :param direct: list all slots directly in the domain of a class, defaults to True
```

Comment:
consider if you really need two boolean parameters here. If both are specified, you raise an error (good). But if neither are specified (which is the default) then this returns an empty list.

if you look at other methods you will see it's more idiomatic to have the method always return direct and to have a single boolean param to determine whether this should be supplemented with induced